src/bsoncxx/builder/core.cpp

```diff
@@ -622,7 +622,9 @@ core& core::open_array() {
 core& core::concatenate(const bsoncxx::document::view& view) {
     if (_impl->is_array()) {
         bson_iter_t iter;
-        bson_iter_init_from_data(&iter, view.data(), view.length());
+        if (!bson_iter_init_from_data(&iter, view.data(), view.length())) {
```

Comment:
Coverity was pointing out that we weren't checking the return value here, so I added the logic that seemed reasonable.